app/app.go

```diff
@@ -484,23 +487,65 @@ func NewAxelarApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
 	// CanWithdrawInvariant invariant.
 	// NOTE: staking module is required if HistoricalEntries param > 0
 	app.mm.SetOrderBeginBlockers(
+		// upgrades should be run first
 		upgradetypes.ModuleName,
 		capabilitytypes.ModuleName,
-		minttypes.ModuleName,
+		crisistypes.ModuleName,
+		govtypes.ModuleName,
+		stakingtypes.ModuleName,
```

Comment:
Wow, so we were never running the begin blocker for those modules? Was that because they didn't used to have a non empty begin blocker?

Reply:
Yes, they are still empty, but new version required the func has all modules

Reply:
Hmm, the order here is different from the [SDK](https://github.com/cosmos/cosmos-sdk/blob/v0.45.1/simapp/app.go#L332) ordering, what's the reason for that? According to the Note there, the staking begin blocker is required if historical entries param is > 0, which is the [case for us](https://axelar-lcd.quickapi.com/swagger/#/Query/StakingParams)!

Also, `authz` module is missing, but doesn't seem like we care about it's functionality rn.

Similarly, does the `NewManager` module order matter? It's different than the [sdk](https://github.com/cosmos/cosmos-sdk/blob/v0.45.1/simapp/app.go#L310).

And according to the release notes, we can now use `SetOrderMigration` too in the future.

Reply:
I believe only the upgrade module order matters, the others do not